### PR TITLE
Program of Binary tree

### PR DESCRIPTION
**Pull request by neha07kumari**
This PR is created for the issue #1477 .

### CODE REVIEW

**Comment by raksha009** [comment]
Plz link yr code with the respective folder. Go through pinned issues and guidlines.

File: Tree/distance_between_two_node_of_a_BT.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+   cout<<distance(root,i,j)<<endl;  
+   
+   //if i=4 ,j=5 then distance between them is 4.
+   //if i=3 ,j=5 then distance between them is 1.
+  
+    return 0;
+}
+
+//TIME COMPLEXITY:O(h) where h is the height of tree 
+//SPACE COMPLEXITY:O(1)
```
Comment (krishna-NIT) [inline Tree/distance_between_two_node_of_a_BT.cpp:97]: Add Sample Test cases

File: Tree/distance_between_two_node_of_a_BT.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    if(root->data == n1 || root->data == n2){
+        return root;
+    }
+
+    Node* left=LCA(root->left,n1,n2);
+    Node* right=LCA(root->right,n1,n2);
+
+    if(left!=NULL && right!=NULL){
```
Comment (krishna-NIT) [inline Tree/distance_between_two_node_of_a_BT.cpp:38]: Add Space in-between " != "
and do this everwhere

File: Tree/distance_between_two_node_of_a_BT.cpp (hunk 1 above)
Comment (neha07kumari) [reply Tree/distance_between_two_node_of_a_BT.cpp:97]: @krishna-NIT  I have added the sample test cases in line 90 & 91.

File: Tree/distance_between_two_node_of_a_BT.cpp (hunk 2 above)
Comment (neha07kumari) [reply Tree/distance_between_two_node_of_a_BT.cpp]: done.

**Comment by neha07kumari** [comment]
> Plz link yr code with the respective folder. Go through pinned issues and guidlines.

@raksha009 Done. Please check it.

**Comment by rudrakshi99** [comment]
Replace `NULL` with `nullptr`.

File: Code/C++/distance_between_two_nodes_of_BT.cpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    root->left = new Node(2);
+    root->right = new Node(3);
+    root->left->left = new Node(4);
+    root->right->right = new Node(5);
+
+   int i,j;
+    cin>>i>>j;
+   cout<<distance(root,i,j)<<endl;  
+   
+   //TEST CASE1:
```
Comment (raksha009) [inline Code/C++/distance_between_two_nodes_of_BT.cpp:90]: Add test cases below yr code.
And add Test cases + Complexities at the end in single multiline comment.

Also, add necessary short comments above the func whereever needed.

Comment (neha07kumari) [reply Code/C++/distance_between_two_nodes_of_BT.cpp:90]: @raksha009  I have added the test cases from line 90-96 & the complexity is in line 101 & 102.

**Review by raksha009 — APPROVED** [review]

**Pull request merged**